src/invocation/ClientConnection.ts

```diff
@@ -66,6 +66,13 @@ class ClientConnection {
             this.socket = netSocket as stream.Duplex;
         }
 
+        netSocket.setTimeout(this.clientNetworkConfig.connectionTimeout)
+        this.socket.on('timeout', (e: any) => {
+            this.logging.warn('ClientConnection',
+                'Could not connect to address ' + Address.encodeToString(this.address) + '. Connection timed out.', e);
+            ready.reject(e);
+        });
+
```

Comment:
Thank you for you effort. Pull requests are always very welcome. I would like to get this merged before next release if you make some little adjustments.

What would be great is if we could have tests to assure

1. - client indeed fails with this particular error message when connection attempt timeouts.
2. - connection attemp does not timeout when `connectionTimeout` is set to 0(unlimited).

You may create a test file in `test` folder, such as `test/ClientConnectionTest`.